### ISSUE TITLE
Provider v0.84.0 crashes when running snowflake_grant_database_role

### ISSUE THREAD

**Issue by adrianhicken** [Snowflake-Labs/terraform-provider-snowflake#2402]
### Terraform CLI and Provider Versions

`Terraform v1.6.4 on darwin_arm64`

Provider version v0.84.0

### Terraform Configuration

```terraform
resource "snowflake_database_role" "database_role" {
  database = "DB_TEST_DB_DEV"
  name     = "TEST_DATABASE_ROLE_02"
}

resource "snowflake_database_role" "parent_database_role" {
  database = "DB_TEST_DB_DEV"
  name     = "TEST_DATABASE_ROLE_01"
}

resource "snowflake_grant_database_role" "g" {
  database_role_name        = "\"DB_TEST_DB_DEV\".\"${snowflake_database_role.database_role.name}\""
  parent_database_role_name = "\"DB_TEST_DB_DEV\".\"${snowflake_database_role.parent_database_role.name}\""
}
```

### Expected Behavior

Database role TEST_DATABASE_ROLE_02 granted to TEST_DATABASE_ROLE_01

### Actual Behavior

```
panic: runtime error: index out of range [1] with length 1

goroutine 64 [running]:
github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/resources.ReadGrantDatabaseRole(0x14000878780, {0x1065e0960?, 0x14001016d00?})
	github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/resources/grant_database_role.go:184 +0x668
github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/resources.CreateGrantDatabaseRole(0x0?, {0x1065e0960?, 0x14001016d00})
	github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/resources/grant_database_role.go:131 +0x758
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x106622bc8?, {0x106622bc8?, 0x14000a30990?}, 0xd?, {0x1065e0960?, 0x14001016d00?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/resource.go:766 +0x134
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x14000b310a0, {0x106622bc8, 0x14000a30990}, 0x1400086eb60, 0x14000878600, {0x1065e0960, 0x14001016d00})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/resource.go:909 +0x86c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x140009fc648, {0x106622bc8?, 0x14000a307e0?}, 0x14000642f50)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/grpc_provider.go:1074 +0xb08
github.com/hashicorp/terraform-plugin-mux/tf5to6server.v5tov6Server.ApplyResourceChange({{0x106637bd8?, 0x140009fc648?}}, {0x106622bc8, 0x14000a307e0}, 0x0?)
	github.com/hashicorp/terraform-plugin-mux@v0.13.0/tf5to6server/tf5to6server.go:37 +0x58
github.com/hashicorp/terraform-plugin-mux/tf6muxserver.(*muxServer).ApplyResourceChange(0x106622c00?, {0x106622bc8?, 0x14000a304e0?}, 0x14000642f00)
	github.com/hashicorp/terraform-plugin-mux@v0.13.0/tf6muxserver/mux_server_ApplyResourceChange.go:36 +0x184
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0x14000bee000, {0x106622bc8?, 0x1400086bcb0?}, 0x140008724d0)
	github.com/hashicorp/terraform-plugin-go@v0.20.0/tfprotov6/tf6server/server.go:857 +0x3c0
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0x106592a40?, 0x14000bee000}, {0x106622bc8, 0x1400086bcb0}, 0x14000878380, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.20.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:503 +0x164
google.golang.org/grpc.(*Server).processUnaryRPC(0x140000cad20, {0x106622bc8, 0x1400086bc20}, {0x1066331e0, 0x14000102ea0}, 0x1400086d8c0, 0x14000bdf2c0, 0x1074aba78, 0x0)
	google.golang.org/grpc@v1.60.0/server.go:1372 +0xb8c
google.golang.org/grpc.(*Server).handleStream(0x140000cad20, {0x1066331e0, 0x14000102ea0}, 0x1400086d8c0)
	google.golang.org/grpc@v1.60.0/server.go:1783 +0xc4c
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.60.0/server.go:1016 +0x5c
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 52
	google.golang.org/grpc@v1.60.0/server.go:1027 +0x138

Error: The terraform-provider-snowflake_v0.84.0 plugin crashed!
```

### Steps to Reproduce

1. `terraform apply`

### How much impact is this issue causing?

Medium

### Additional Information

```
2024-01-23T12:59:52.864Z [DEBUG] provider.terraform-provider-snowflake_v0.84.0: 2024/01/23 12:59:52 [DEBUG] sql-conn-exec: [query GRANT DATABASE ROLE "DB_TEST_DB_DEV"."TEST_DATABASE_ROLE_02" TO DATABASE ROLE "DB_TEST_DB_DEV"."TEST_DATABASE_ROLE_01" err <nil> duration 224.328125ms args {}] ()
2024-01-23T12:59:53.096Z [DEBUG] provider.terraform-provider-snowflake_v0.84.0: 2024/01/23 12:59:53 [DEBUG] sql-conn-query: [query SHOW GRANTS OF DATABASE ROLE "DB_TEST_DB_DEV"."TEST_DATABASE_ROLE_02" err <nil> duration 231.756667ms args {}] ()
2024-01-23T12:59:53.099Z [DEBUG] provider.terraform-provider-snowflake_v0.84.0: panic: runtime error: index out of range [1] with length 1
```

**Comment by sysadmin-exe** [comment]
Same with the latest 0.84.1

**Comment by sfc-gh-asawicki** [comment]
Hey @adrianhicken. Thanks for reporting the issue.

As this is a new resource we will analyze it shortly.

**Comment by monti-python** [comment]
Sames issue with the latest version 0.85.0

**Comment by davidlee-ca** [comment]
+1 on this - using 0.85.0, still crashing the same way.

**Comment by sfc-gh-jcieslak** [comment]
Hi all, I'm currently working on it. The fix should be merged and released soon as a bugfix version or next week as a regular minor release.

**Comment by sfc-gh-jcieslak** [comment]
Hey @adrianhicken. We have released the fix as part of [v0.87.0](https://github.com/Snowflake-Labs/terraform-provider-snowflake/releases/tag/v0.87.0) release. Please follow the [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/v0.87.0/MIGRATION_GUIDE.md#v0860--v0870) during the update. Please confirm that the issue is resolved in the newest version. Thanks!
